public/app/features/explore/QueryLibrary/QueryTemplatesTable/ActionsCell.tsx

```diff
@@ -1,16 +1,67 @@
 import React from 'react';
 
+import { reportInteraction, getAppEvents } from '@grafana/runtime';
 import { DataQuery } from '@grafana/schema';
+import { IconButton } from '@grafana/ui';
+import { notifyApp } from 'app/core/actions';
+import { createSuccessNotification } from 'app/core/copy/appNotification';
+import { t } from 'app/core/internationalization';
+import { useDeleteQueryTemplateMutation } from 'app/features/query-library';
+import { dispatch } from 'app/store/store';
+import { ShowConfirmModalEvent } from 'app/types/events';
 
 import ExploreRunQueryButton from '../../ExploreRunQueryButton';
 
+import { useQueryLibraryListStyles } from './styles';
+
 interface ActionsCellProps {
+  queryUid?: string;
   query?: DataQuery;
   rootDatasourceUid?: string;
 }
 
-function ActionsCell({ query, rootDatasourceUid }: ActionsCellProps) {
-  return <ExploreRunQueryButton queries={query ? [query] : []} rootDatasourceUid={rootDatasourceUid} />;
+function ActionsCell({ query, rootDatasourceUid, queryUid }: ActionsCellProps) {
+  const [deleteQueryTemplate] = useDeleteQueryTemplateMutation();
+  const styles = useQueryLibraryListStyles();
+
+  const onDeleteQuery = (queryUid: string) => {
+    const performDelete = (queryUid: string) => {
+      deleteQueryTemplate({ uid: queryUid });
+      dispatch(notifyApp(createSuccessNotification(t('explore.query-library.query-deleted', 'Query deleted'))));
+      reportInteraction('grafana_explore_query_library_deleted');
+    };
+
+    getAppEvents().publish(
+      new ShowConfirmModalEvent({
+        title: t('explore.query-library.delete-query-title', 'Delete query'),
+        text: t(
+          'explore.query-library.delete-query-text',
+          "You're about to remove this query from the query library. This action cannot be undone. Do you want to continue?"
+        ),
+        yesText: t('query-library.delete-query-button', 'Delete query'),
+        icon: 'trash-alt',
+        onConfirm: () => performDelete(queryUid),
+      })
+    );
+  };
+
+  return (
+    <div className={styles.cell}>
+      <IconButton
```

Comment:
It seemed to me a bit off that it was in front of the action button to use the template, would expect it on the right side of it. Not sure if that is some convention I am used to or not right now but I have the feeling that if there are multiple buttons per row in some list destructive actions tend to be the last of the actions.

Reply:
I think this will be more clear when we have more actions. I think it's lining up all the actions possible and then having run query as the "main" action on the right like we see elsewhere. We can revisit it, but I'm going to leave this as is now to match the prototype. 

![Screenshot 2024-06-20 at 12 06 58 PM](https://github.com/grafana/grafana/assets/1533128/bf5f9f75-dd47-458a-82da-71f1c085dc43)